### ISSUE TITLE
Electrum export/import derives wrong script type

### ISSUE THREAD

**Issue by zanepocock** [cryptoadvance/specter-desktop#1490]
I have a 2-of-3 multisig segwit wallet. Derivation m/48h/0h/0h/2h for all 3, with xpubs as the public keys.

On Specter this arrangement uses bech32 addresses, but when I import into Electrum from Settings>Export>Create Electrum Wallet, Electrum doesn't seem to recognize the script descriptors and derives P2SH addresses instead. Would it be better for this file to export Zpubs?

**Comment by zanepocock** [comment]
For any onlookers, I ran https://github.com/jlopp/xpub-converter locally to replace the xpubs with their respective Zpubs in the .backup file. Then imported as usual into Electrum. Not very difficult but it would be great if you didn't have to take the extra step.

**Comment by k9ert** [comment]
@moneymanolis can you qualify this? I'm not sure why this changed now. Might be a change on electrum-side. The biggest problem here is that it's a moving target: The other side might not have an official document how it's suppose to look like in order to work.

**Comment by zanepocock** [comment]
Should've added version in case it's on the Electrum side: Electrum V 4.1.5

**Comment by k9ert** [comment]
I'd love to get a specification before i do any fix as otherwise, this is still a moving target and could break any time. Maybe you can create an issue on the electrum-tracker and we get someone who can give a qualified answer why this doesn't work on electrum and give a spec which hopefully work longer if implemented correctly.

So i see that rather as a n electrum-issue. If i fix it here for `v4.1.5` it might even break for older versions. We have to clarify before.

**Comment by zanepocock** [comment]
Thanks @k9ert opened an issue here: https://github.com/spesmilo/electrum/issues/7596

**Comment by moneymanolis** [comment]
> @moneymanolis can you qualify this?

I checked the code in Specter for the Electrum export (`get_electrum_file()` in `wallet.py`): I suppose since `key.original` is returned, that will be xpubs without special version bytes (hence no Zpubs), since BIP48 isn't considered in SLIP132 (BIP48 is pretty new). I think Electrum is not inferring the right script type from the derivation path alone (as Specter does for example). Converting the version bytes probably lets Electrum come to the correct script type (P2WSH). I'd agree with @k9ert that we'd need a specification from Electrum. 

For the time being, converting version bytes as you did @zanepocock should do the trick. I am closing this for now. @zanepocock, feel free to open a feature request once you have a specification from Electrum.

**Comment by zanepocock** [comment]
Hi @k9ert @moneymanolis just coming back to close the loop on this [after hearing back from Electrum](https://github.com/spesmilo/electrum/issues/7596#issuecomment-1015613512).

TL;DR they believe that they haven't changed anything "in years" that would've affected this integration. As for future plans:

> We intend to deprecate y/zpubs and use some form of output script descriptors (although their base form is not a good fit to represent a single cosigner unfortunately).

**Comment by k9ert** [comment]
pinging here @relativisticelectron as he's also quite active on electrum-integration.
Other than that, thanks you for your work on this.
